tests/preflight/preflight_test.go

```diff
@@ -31,222 +28,258 @@ var _ = Describe("Preflight Tests", func() {
+				assertSuccessfulPreflightChecks(inputFlags, cmdOut.Out)
+			})
+
+			It("Should fail preflight check if namespace flag is provided with zero value", func() {
+				var output []byte
+				args := []string{"run", storageClassFlag, defaultTestStorageClass, namespaceFlag, ""}
+				cmd := exec.Command("./tvk-preflight", args...)
+				output, err = cmd.CombinedOutput()
+				Expect(err).To(BeNil())
+
+				Expect(string(output)).To(ContainSubstring("Preflight check for DNS resolution failed :: " +
+					"an empty namespace may not be set during creation"))
+				Expect(string(output)).To(ContainSubstring("Preflight check for volume snapshot and restore failed :: " +
+					"an empty namespace may not be set during creation"))
 			})
 		})
 
 		Context("Preflight run command, kubeconfig flag test cases", func() {
 			It("Should perform preflight checks when a kubeconfig file is specified", func() {
-				var (
-					byteData         []byte
-					testKubeConfFile *os.File
-					kubeConfFile     *os.File
-				)
+				var byteData []byte
 				inputFlags := make(map[string]string)
 				copyMap(flagsMap, inputFlags)
-				testKubeConfFile, err = os.OpenFile(preflightKubeConf, os.O_CREATE|os.O_WRONLY, filePermission)
+				_, err = os.Create(preflightKubeConf)
+				Expect(err).To(BeNil())
+				defer func() {
+					err = os.Remove(preflightKubeConf)
+					Expect(err).To(BeNil())
+				}()
+				byteData, err = ioutil.ReadFile(kubeConfPath)
 				Expect(err).To(BeNil())
-				defer testKubeConfFile.Close()
-				kubeConfFile, err = os.OpenFile(kubeConfPath, os.O_RDONLY, filePermission)
+				err = ioutil.WriteFile(preflightKubeConf, byteData, filePermission)
 				Expect(err).To(BeNil())
-				defer kubeConfFile.Close()
-				copyFile(kubeConfFile, testKubeConfFile)
-				inputFlags[kubeconfigFlag] = path.Join(".", testKubeConfFile.Name())
-				runPreflightChecks(inputFlags)
-				byteData, err = getLogFileData(preflightLogFilePrefix)
+
+				inputFlags[kubeconfigFlag] = path.Join(".", preflightKubeConf)
+				cmdOut, err = runPreflightChecks(inputFlags)
 				Expect(err).To(BeNil())
-				outputLogData := string(byteData)
 
-				Expect(outputLogData).To(ContainSubstring(
-					fmt.Sprintf("Using kubeconfig file path - %s", path.Join(".",
-						testKubeConfFile.Name()))))
+				Expect(cmdOut.Out).To(ContainSubstring(
+					fmt.Sprintf("Using kubeconfig file path - %s", path.Join(".", preflightKubeConf))))
+				assertSuccessfulPreflightChecks(inputFlags, cmdOut.Out)
+			})
 
-				err = os.Remove(testKubeConfFile.Name())
+			It("Should fail preflight execution if invalid kubeconfig file is provided", func() {
+				_, err = os.Create(invalidKubeConfFilename)
 				Expect(err).To(BeNil())
+				err = ioutil.WriteFile(invalidKubeConfFilename, []byte(invalidKubeConfFileData), filePermission)
+				Expect(err).To(BeNil())
+				inputFlags := make(map[string]string)
+				copyMap(flagsMap, inputFlags)
+				inputFlags[kubeconfigFlag] = invalidStorageClassName
```

Comment:
why  `kubeconfigFlag=invalidStorageClassName`? this does not seem  invalid kubeconfig test-case